tests/test_dulwich.py

```diff
@@ -26,3 +30,21 @@ def test_dulwich_github_compat(mocker: MockerFixture, algorithm: bytes):
     strings = iter((b"ssh-rsa", key_data))
     packet.get_string = lambda: next(strings)
     _process_public_key_ok_gh(auth, None, None, packet)
+
+
+def test_dulwich_get_ref(tmp_dir: TmpDir, scm: Git):
+    backend = DulwichBackend(tmp_dir)
+    tmp_dir.gen("foo", "foo")
+    scm.add_commit("foo", message="foo")
+
+    scm.set_ref("refs/exp/base", "refs/heads/master")
+
+    custom_ref = backend.get_ref("refs/exp/base", follow=False)
+    assert str(custom_ref).rstrip("\n") == "refs/heads/master"
```

Comment:
This strip shouldn't be necessary, if `get_ref` is returning a string with whitespace on either end the test should fail.

(I think the newline is probably because of the missing `symbolic` in `set_ref`?)